flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/DataStream.scala

```diff
@@ -144,6 +145,43 @@ class DataStream[T](stream: JavaStream[T]) {
     this
   }
 
+
+  /**
+   * Returns the minimum resource of this operation.
+   */
+  def getMinResource: ResourceSpec = stream.getMinResource()
+
+  /**
+   * Returns the maximum resource of this operation.
+   */
+  def getMaxResource: ResourceSpec = stream.getMaxResource()
+
+  /**
+   * Sets the minimum and maximum resources of this operation.
+   */
+  def setResource(minResource: ResourceSpec, maxResource: ResourceSpec): DataStream[T] = {
+    stream match {
+      case ds: SingleOutputStreamOperator[T] => ds.setResource(minResource, maxResource)
+      case _ =>
+        throw new UnsupportedOperationException(
+          "Operator " + stream + " cannot set the resource.")
+    }
+    this
+  }
+
+  /**
+   * Sets the resource of this operation.
+   */
+  def setResource(resource: ResourceSpec): DataStream[T] = {
```

Comment:
The more Scala-like way of writing this would be
```scala
def resource_=(resource: ResourceSpec): Unit = {
    ...
}
```
That way you can write Scala code like this (and it will translate to the setter call):
```scala
val stream: DataStream[MyType] = ...
stream.resource = new Resource(a, b, c, d)
```
(We have missed that in most places in Flink so far, but it may make sense to start picking up the proper Scala style for new changed)